src/functions.php

```diff
@@ -79,6 +80,38 @@ function generate_index_name($document)
 }
 
 /**
+ * Return whether the error code and message indicate a duplicate key error.
+ *
+ * @internal
+ * @param integer $code
+ * @param string  $message
+ * @return boolean
+ */
+function is_duplicate_key_error($code, $message)
```

Comment:
Based on [`_is_duplicate_key_error()`](https://github.com/mongodb/mongo-c-driver/blob/1.8.1/src/mongoc/mongoc-write-command.c#L62) from libmongoc.

Reply:
Note, sine PHPC 1.5.0 and PHPLIB 1.4.0 dropped support for MongoDB versions before 3.0, we need only consider `11000`. The other error codes all pertain to 2.6 and earlier servers.